File: app/src/main/java/com/quibbler/sevenmusic/adapter/found/FoundSingerLibRecommendAdapter.java
package com.quibbler.sevenmusic.adapter.found;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.quibbler.sevenmusic.MusicApplication;
import com.quibbler.sevenmusic.R;

import com.quibbler.sevenmusic.activity.found.SingerActivity;
import com.quibbler.sevenmusic.bean.jsonbean.found.FoundSingerInfo;
import com.quibbler.sevenmusic.presenter.ImageDownloadPresenter;


import java.util.List;

public class FoundSingerLibRecommendAdapter extends RecyclerBaseAdapter<FoundSingerInfo, FoundSingerLibRecommendAdapter.SingerTopViewHolder> {
    //实际使用的数据源list
    private List<FoundSingerInfo> mFoundSingerInfoList = mSourceList;

    static class SingerTopViewHolder extends RecyclerView.ViewHolder {
        View mView;
        ImageView mImageView;
        TextView mTextView;

        public SingerTopViewHolder(View view) {
            super(view);
            mView = view;
            mImageView = view.findViewById(R.id.found_list_item_top_singer_iv);
            mTextView = view.findViewById(R.id.found_list_item_singer_top_tv_name);
        }
    }

    public FoundSingerLibRecommendAdapter(List<FoundSingerInfo> foundSingerInfoList) {
        super(foundSingerInfoList);
    }

    @NonNull
    @Override
    public SingerTopViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.found_list_item_top_singer, parent, false);
        SingerTopViewHolder viewHolder = new SingerTopViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull SingerTopViewHolder holder, int position) {
        super.onBindViewHolder(holder, position);

        FoundSingerInfo foundSingerInfo = mFoundSingerInfoList.get(position);
        holder.mTextView.setText(foundSingerInfo.getName());

        ImageDownloadPresenter.getInstance().with(holder.mImageView.getContext())
                .load(foundSingerInfo.getPicUrl())
                .imageStyle(ImageDownloadPresenter.STYLE_CIRCLE)
                .into(holder.mImageView);

        holder.mView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(holder.mView.getContext(), SingerActivity.class);
                intent.putExtra("id", foundSingerInfo.getId());
                holder.mView.getContext().startActivity(intent);
            }
        });
    }

}

File: app/src/main/java/com/quibbler/sevenmusic/adapter/found/FoundSingerLibRecommendAdapter.kt
package com.quibbler.sevenmusic.adapter.found

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.quibbler.sevenmusic.R
import com.quibbler.sevenmusic.activity.found.SingerActivity
import com.quibbler.sevenmusic.adapter.found.FoundSingerLibRecommendAdapter.SingerTopViewHolder
import com.quibbler.sevenmusic.bean.jsonbean.found.FoundSingerInfo
import com.quibbler.sevenmusic.presenter.ImageDownloadPresenter

class FoundSingerLibRecommendAdapter(foundSingerInfoList: MutableList<FoundSingerInfo?>?) :
    RecyclerBaseAdapter<FoundSingerInfo?, SingerTopViewHolder?>(foundSingerInfoList) {
    //实际使用的数据源list
    private val mFoundSingerInfoList: MutableList<FoundSingerInfo> = mSourceList

    internal class SingerTopViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var mView: View
        var mImageView: ImageView
        var mTextView: TextView

        init {
            mView = view
            mImageView = view.findViewById<ImageView>(R.id.found_list_item_top_singer_iv)
            mTextView = view.findViewById<TextView>(R.id.found_list_item_singer_top_tv_name)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SingerTopViewHolder {
        val view = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.found_list_item_top_singer, parent, false)
        val viewHolder = SingerTopViewHolder(view)
        return viewHolder
    }

    override fun onBindViewHolder(holder: SingerTopViewHolder, position: Int) {
        super.onBindViewHolder(holder, position)

        val foundSingerInfo = mFoundSingerInfoList.get(position)
        holder.mTextView.setText(foundSingerInfo.getName())

        ImageDownloadPresenter.Companion.getInstance().with(holder.mImageView.getContext())
            .load(foundSingerInfo.getPicUrl())
            .imageStyle(ImageDownloadPresenter.Companion.STYLE_CIRCLE)
            .into(holder.mImageView)

        holder.mView.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                val intent = Intent(holder.mView.getContext(), SingerActivity::class.java)
                intent.putExtra("id", foundSingerInfo.getId())
                holder.mView.getContext().startActivity(intent)
            }
        })
    }
}
